how-to-guides/hello-neptune/notebooks/hello_neptune.ipynb

```diff
@@ -151,24 +138,55 @@
     "    capture_hardware_metrics=True,\n",
     "    capture_stderr=True,\n",
     "    capture_stdout=True,\n",
-    "    source_files=\"hello_neptune.ipynb\",  # does not work in Colab\n",
+    "    source_files=\"hello_neptune.ipynb\",  # Doesn't work in Colab\n",
     "    tags=[\"quickstart\", \"notebook\"],\n",
-    "    dependencies=\"requirements.txt\",  # pass the path to the requirements.txt file\n",
+    "    dependencies=\"requirements.txt\", # You can pass the path to your dependencies file or omit this parameter \n",
     ")"
    ]
   },
   {
-   "attachments": {},
    "cell_type": "markdown",
-   "metadata": {
-    "tags": [
-     "comment"
-    ]
-   },
+   "metadata": {},
+   "source": [
+    "### Option B: Use your own project\n",
+    "\n",
+    "To use your own project, pass your credentials to the `init_run()` function.\n",
+    "\n",
+    "To find your credentials:\n",
+    "\n",
+    "1. [Log in to Neptune](https://app.neptune.ai/).\n",
+    "1. In the bottom-left corner, expand your user menu and select **Get your API token**.\n",
+    "1. In the project view, in the top-right corner, select the menu, and then **Edit project details**."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
    "source": [
-    "**To open the run in the Neptune web app, click the link that appeared in the cell output.**\n",
+    "import neptune\n",
+    "from getpass import getpass\n",
     "\n",
-    "You'll use the `run` object you just created to log metadata. You'll see the metadata appear in the app."
+    "run = neptune.init_run(\n",
+    "    project=\"workspace-name/project-name\",  # For example, \"ml-team/nlu\"\n",
+    "    api_token=getpass(\"Enter your Neptune API token: \"),\n",
+    "    capture_hardware_metrics=True,\n",
+    "    capture_stderr=True,\n",
+    "    capture_stdout=True,\n",
+    "    source_files=\"hello_neptune.ipynb\",  # Doesn't work in Colab\n",
+    "    tags=[\"quickstart\", \"notebook\"],\n",
+    "    dependencies=\"requirements.txt\",  # You can pass the path to your dependencies file or omit this parameter\n",
+    ")"
```

Comment:
Our tests run notebooks end-to-end using `ipython`, so we need to ensure that any code cell can be run successfully as-is.
Also, `getpass()` does not play along well with `ipython` in non-interactive mode. 

I'd suggest having "Option A" as using your own project and converting the code cell to a markdown (formatted as code).

This will let users copy-paste the code from option A in the code cell for option B (public project) if they want to use their own project, serving two benefits:

1. Our tests run fine end-to-end
2. Users don't create an anonymous run in option A, and then leave a hanging run by creating another one in option B

Reply:
Ah... This does ring a bell. I did consider point 2 but concluded that it's not a big deal, as this should not happen that often -- and even when it does, it just sends a spam run to `quickstarts` that will soon be cleaned up. Sparing users the copy-pasting would definitely be worth it.

I appreciate the need to keep things testable, but from a UX perspective, I really got to like the idea of having parallel runnable code blocks. It's clearer from a presentation perspective and better supports using your own project.

Would there be any way to work around this in our test setup? Or maybe we could exempt Hello Neptune from tests, since it's a very simple example with hardly any extra dependencies? I'd just like to try exploring some options before abandoning this idea entirely.

Reply:
There might be a workaround: https://neptune-labs.slack.com/archives/C01H232MYG5/p1712733838010849
It is not pretty, and I'll need a couple of days to check if this is even feasible without any user-facing changes. If it is, I'll need more time to adapt our test setup to use it.

I wouldn't exempt `hello_neptune` from tests as it does have an external dependency, `mnist`. 

Also, if we adopt the A/B approach when initializing a run, we'll need to replicate it when reinitializing the run later in the notebook while fetching logged metadata from Neptune.